src/main/java/org/broadinstitute/consent/http/service/LibraryCardService.java

```diff
@@ -112,11 +114,15 @@ public void removeDaaFromLibraryCard(Integer libraryCardId, Integer daaId) {
     libraryCardDAO.deleteLibraryCardDaaRelation(libraryCardId, daaId);
   }
 
-  public List<LibraryCard> addDaaToUserLibraryCardByInstitution(User user, Integer institutionId, Integer daaId) {
+  public List<LibraryCard> addDaaToUserLibraryCardByInstitution(User user, User signingOfficial, Integer daaId) {
     List<LibraryCard> libraryCards = findLibraryCardsByUserId(user.getUserId());
     List<LibraryCard> matchingLibraryCards = libraryCards.stream()
-        .filter(card -> Objects.equals(card.getInstitutionId(), institutionId))
-        .toList();
+        .filter(card -> Objects.equals(card.getInstitutionId(), signingOfficial.getInstitutionId()))
+        .collect(Collectors.toCollection(ArrayList::new));
+    if (matchingLibraryCards.isEmpty()) {
+      LibraryCard lc = createLibraryCardForSigningOfficial(user, signingOfficial);
+      matchingLibraryCards.add(lc);
+    }
```

Comment:
I'm finding the overall logic for this method a little convoluted. We do the following things:
1. Find all user LCs
2. Filter them on an institution
3. If none, create new LC
4. For each LC, add DAA to it
5. Return the filtered LCs without having updated them with the new DAAs

What would make more sense to me is the following logical process
1. Find user LCs by user and institution ID (likely a new dao call)
  a. If none, then create new LC
2. Issue a call to a new bulk insert DAO call that adds DAA association for all instances of user id and institution id. See, for example, [this method for bulk inserting](https://github.com/DataBiosphere/consent/blob/8305d4748f6c6589fc68723de759c871c9a2283e/src/main/java/org/broadinstitute/consent/http/db/UserPropertyDAO.java#L29) user properties
3. Return a fresh DAO call of all LCs by user id and institution id so that the DAA ids are populated on the LCs.